ablySpec/RealtimeClientChannel.swift

```diff
@@ -980,6 +980,124 @@ class RealtimeClientChannel: QuickSpec {
                     expect(restChannelHistoryMethodWasCalled).to(beTrue())
                 }
 
+                // RTL10b
+                pending("supports the param untilAttach") {
+
+                    it("should be false as default") {
+                        let query = ARTRealtimeHistoryQuery()
+                        expect(query.untilAttach).to(equal(false))
+                    }
+
+                    struct CaseTest {
+                        let untilAttach: Bool
+                        let expected: Int
+                    }
+
+                    let cases = [CaseTest(untilAttach: true, expected: 1),
+                                 CaseTest(untilAttach: false, expected: 2)]
+
+                    for caseItem in cases {
+                        it("where value is \(caseItem.untilAttach)") {
+                            let client = ARTRealtime(options: AblyTests.commonAppSetup())
+                            defer { client.close() }
+
+                            let mockExecutor = MockHTTPExecutor()
+                            client.rest.httpExecutor = mockExecutor
+
+                            let channel = client.channels.get("test")
+
+                            let query = ARTRealtimeHistoryQuery()
+                            query.untilAttach = caseItem.untilAttach
+
+                            channel.publish(nil, data: "preMessage")
+
+                            expect(channel.state).toEventually(equal(ARTRealtimeChannelState.Attached), timeout: testTimeout)
+
+                            waitUntil(timeout: testTimeout) { done in
+                                channel.publish(nil, data: "postMessage") { errorInfo in
+                                    expect(errorInfo).to(beNil())
+                                    done()
+                                }
+                            }
+
+                            waitUntil(timeout: testTimeout) { done in
+                                try! channel.history { result, _ in
+                                    expect(result!.items.count).to(equal(caseItem.expected))
```

Comment:
This is basically testing the same as the `it` below, only at a smaller scale. To simplify, maybe we can remove it, and just test the `from_serial` in the URL query?